### PR TITLE
Apply small fixes to CI and Scala config

### PR DESCRIPTION
**Pull request by PawelLipski**

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -20,36 +20,31 @@ jobs:
...
           command: ./scripts/run-pre-build-checks
       - restore_cache:
-          # Adding `-v4` suffix to enable easy cache invalidation when needed (just change to `-v<next-integer>`)
+          # Adding `-v<N>` suffix to enable easy cache invalidation when needed:
+          # just change to `-v<N+1>` (remember to also update `save_cache` key).
           keys:
             # First, try finding a cache entry with the same set of libraries
             # that also comes from the same branch (so as to make the best use of Gradle compilation cache).
-            - gradle-deps-v4-{{ checksum "build.gradle" }}-{{ .Branch }}
+            - gradle-deps-v5-{{ checksum "gradle/libs.versions.toml" }}-{{ .Branch }}
```
Comment (PawelLipski) [inline .circleci/config.yml:31]: Cache key computed out of library versions that actually matter for the sake of caching

File: scripts/run-pre-build-checks [hunk 2]
```diff
@@ -32,3 +32,9 @@ prohibit-tab-character
 prohibit-trailing-whitespace
 
 validate-icons
+if [[ $0 != *.git/hooks/pre-commit ]]; then
```
Comment (mkondratek) [inline scripts/run-pre-build-checks:35]: good idea 👍

File: gradle/libs.versions.toml [hunk 3]
```diff
@@ -20,6 +20,7 @@ lombok = "org.projectlombok:lombok:1.18.24"
 powerMock-apiMockito2 = { module = "org.powermock:powermock-api-mockito2", version.ref = "powerMock" }
 powerMock-moduleJunit4 = { module = "org.powermock:powermock-module-junit4", version.ref = "powerMock" }
 reflections = "org.reflections:reflections:0.10.2"
+scala-library = "org.scala-lang:scala-library:2.13.8"
```
Comment (mkondratek) [inline gradle/libs.versions.toml:23]: 1 patch too late :trollface:

**Review by mkondratek — APPROVED** [review]
lgtm

**Pull request merged**